File: apps/web/src/types/pagination.ts
type PaginatedResponse<T> = {
  items: T[];
  total: number;
  page: number;
  pageSize: number;
  totalPages: number;
};

type PaginatedApiResponse<T> = {
  items: T[];
  page: number;
  page_size: number;
  total: number;
  total_pages: number;
};

type PaginationParams = {
  skip: number;
  take: number;
};

export type { PaginatedApiResponse, PaginatedResponse, PaginationParams };

File: apps/api/app/core/pagination.py
def paginate(items: list, total: int, skip: int, take: int) -> dict:
    total_pages = (total + take - 1) // take if total else 0
    page = (skip // take) + 1 if total else 1

    return {
        "items": items,
        "total": total,
        "page": page,
        "page_size": take,
        "total_pages": total_pages,
    }